From 88519aed6c7f305f7f2319e335c1421137df7ce3 Mon Sep 17 00:00:00 2001
From: Ron Nazarov <ron@noisytoot.org>
Date: Mon, 23 Mar 2026 17:04:03 +0000
Subject: [PATCH] mb/supermicro/x11-lga1151-series: Enable SATA hotplug

Before this patch, hotplugging only worked to replace drives (if you
tried to plug a drive into a SATA port that no drive was plugged in to
at boot, it wouldn't be detected) and you'd have to manually rescan
the bus (echo "- - -" > /sys/class/scsi_host/host*/scan) to make
plugs/unplugs get detected by the operating system.

Now, hotplugging works for all ports (tested and working on Supermicro
X11SSH-LN4F) and there's no need to manually rescan (it sometimes
takes a few seconds for unplugs to be detected, but plugs are detected
instantly).

Change-Id: Id978a047697795ea657048fb6dc6665736c293f9
Signed-off-by: Ron Nazarov <ron@noisytoot.org>
---
 .../supermicro/x11-lga1151-series/devicetree.cb        | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/mainboard/supermicro/x11-lga1151-series/devicetree.cb b/src/mainboard/supermicro/x11-lga1151-series/devicetree.cb
index fbf896c6ae..d25288420f 100644
--- a/src/mainboard/supermicro/x11-lga1151-series/devicetree.cb
+++ b/src/mainboard/supermicro/x11-lga1151-series/devicetree.cb
@@ -28,6 +28,16 @@ chip soc/intel/skylake
 				[6] = 1,
 				[7] = 1,
 			}"
+			register "SataPortsHotPlug" = "{
+				[0] = 1,
+				[1] = 1,
+				[2] = 1,
+				[3] = 1,
+				[4] = 1,
+				[5] = 1,
+				[6] = 1,
+				[7] = 1,
+			}"
 		end
 		device ref lpc_espi on
 			register "serirq_mode" = "SERIRQ_CONTINUOUS"
-- 
2.52.0

